Sources/Plasma/PubUtilLib/plGImage/plPNG.cpp

```diff
@@ -226,27 +229,57 @@ bool plPNG::IWrite(plMipmap* source, hsStream* outStream)
             row_ptrs[i] = (png_bytep)srcp + (i * stride);
         }
 
+        // Write image data
         png_write_image(png_ptr, row_ptrs);
+
+        // Prepare Textual Metadata
+        char time_string[26]; // RFC-1123: "Day, DD-Mon-YYYY hh:mm:ss";
+        time_t rawtime = time(nullptr);
+        strftime(time_string, sizeof(time_string), "%a, %d-%b-%Y %T", gmtime(&rawtime));
+        std::multimap<ST::string, ST::string> standard_fields;    
+        standard_fields.emplace("Software", plProduct::ProductString());
+        standard_fields.emplace("Creation Time", time_string);
+
+        png_text* text = new png_text[standard_fields.size() + textFields.size()];
+        int num_txtfields = 0;
+        for (auto it = standard_fields.begin(); it != standard_fields.end(); it++, num_txtfields++) {
+            // The PNG specification requires Latin-1 in these fields
+            text[num_txtfields].key = const_cast<png_charp>(it->first.left(79).to_latin_1().data());
+            text[num_txtfields].text = const_cast<png_charp>(it->second.to_latin_1().data());
```

Comment:
Thanks, I had come to guess that's what I was doing wrong here, and was hoping I could avoid the extra mess of storing and freeing them all in a third container.